internal/generate/tags/templates/v2/wait_tags_propagated_body.gtpl

```diff
@@ -41,3 +23,29 @@ func {{ .WaitTagsPropagatedFunc }}(ctx context.Context, conn {{ .ClientType }},
 
 	return tfresource.WaitUntil(ctx, {{ .WaitTimeout }}, checkFunc, opts)
 }
+
+// checkFunc returns a function that checks if the tags are propagated.
+func checkFunc(ctx context.Context, conn {{ .ClientType }}, tags tftags.KeyValueTags, id string, optFns ...func(*{{ .AWSService }}.Options)) func() (bool, error) {
```

Comment:
`checkFunc` as a name seems a big vague and could well conflict with contributor-written code. How about something like `checkTagsPropagated`?